pennylane/transforms/batch_input.py

```diff
@@ -73,25 +72,39 @@ def circuit(inputs, weights):
     array([0.46230079, 0.73971315, 0.95666004, 0.5355225 , 0.66180948,
             0.44519553, 0.93874261, 0.9483197 , 0.78737918, 0.90866411])>
     """
-    parameters = tape.get_parameters(trainable_only=False)
-
     argnum = tuple(argnum) if isinstance(argnum, (list, tuple)) else (int(argnum),)
 
-    non_trainable, trainable = [], []
-    for idx, param in enumerate(parameters):
-        if idx in argnum:
-            non_trainable.append(param)
-        else:
-            trainable.append(param)
+    trainable_params = tape.get_parameters()
+    all_parameters = sum(
+        (op.parameters if op.parameters != [] else [None] for op in tape.operations), []
+    )
```

Comment:
I think this approach does not quite work out yet, see below at the "batch-composition" stage of the function.
If I see this right, it might be possible to use `tape.get_parameters(trainable_only=False)` here.

Reply:
As mentioned in the PR definition:

> Cannot use tape.get_parameters(trainable_only=False) to get all parameters because it skips non-parametric operations. Consequently the argnum indices might point to the wrong operator.

Reply:
I think this directly relates to the other thread with the `Hadamard`-containing example. :)